enhancements/machine-config/admin-defined-node-disruption-policy.md

```diff
@@ -0,0 +1,311 @@
+* Have users be able to easily see existing cluster non-disruptive update cases
+
+### Non-Goals
+
+* Have NodeDisruptionPolicy apply to non-MCO driven changes (e.g. SRIOV can still reboot nodes)
+* Remove existing non-disruptive update paths (the user will be able to override cluster defaults)
+* Design for image-based updates (live apply and bootc, will be considered in the future)
+* Have the MCO validate whether a change can be successfully applied with the given NodeDisruptionPolicy (i.e. it is up to the responsibility of the user to ensure the correctness of their defined actions)
+
+## Proposal
+
+Create NodeDisruptionPolicySpec and NodeDisruptionPolicyStatus in the machineconfigurations.operator.openshift.io CRD, allowing specify user-provided NodeDisruptionPolicy, and view current version cluster defaults, as follows:
+
+```console
+apiVersion: operator.openshift.io/v1
+kind: MachineConfiguration
+spec:
+  nodeDisruptionPolicySpec:
+    userPolicies:
+      - type: file
+        value: "/etc/my-file"
+        actions:
+          - type: reload
+            reload: my-service
+          - type: daemon-reload
+    clusterDefaultPolicies:
+      - type: file
+        value: /etc/mco/internal-registry-pull-secret.json
+        actions:
+          - type: none
+      - type: file
+        value: /var/lib/kubelet/config.json
+        actions:
+          - type: none
+      - type: file
+        value: /etc/machine-config-daemon/no-reboot/containers-gpg.pub
+        actions:
+          - type: reload
+            reload: crio.service
+      - type: file
+        value: /etc/containers/policy.json
+        actions:
+          - type: reload
+            reload: crio.service
+      - type: file
+        value: /etc/containers/registries.conf
+        actions:
+          - type: special
+status:
+  nodeDisruptionPolicyStatus:
+    clusterPolicies:
+      - type: file
+        value: "/etc/my-file"
+        actions:
+          - type: reload
+            reload: my-service
+          - type: daemon-reload
+      - type: file
+        value: /etc/mco/internal-registry-pull-secret.json
+        actions:
+          - type: none
+      - type: file
+        value: /var/lib/kubelet/config.json
+        actions:
+          - type: none
+      - type: file
+        value: /etc/machine-config-daemon/no-reboot/containers-gpg.pub
+        actions:
+          - type: reload
+            reload: crio.service
+      - type: file
+        value: /etc/containers/policy.json
+        actions:
+          - type: reload
+            reload: crio.service
+      - type: file
+        value: /etc/containers/registries.conf
+    validation: Success
+```
+
+Which, when applied, will define actions for all future changes to these MachineConfig fields based on the diff of the update (rendered MachineConfigs).
+
+For this initial implementation we will support MachineConfig changes to:
+ - Files
+ - Units
+ - kernelArguments
```

Comment:
how are we envisioning  applying kargs without performing reboot?

Reply:
Removed that from the list in the API and here